src/libsystemd/sd-path/sd-path.c

```diff
@@ -98,10 +99,17 @@ static int from_user_dir(const char *field, char **buffer, const char **ret) {
         assert(buffer);
         assert(ret);
 
-        r = from_home_dir(NULL, ".config/user-dirs.dirs", &b, &fn);
+        r = from_home_dir("XDG_CONFIG_HOME", ".config", &b, &c);
         if (r < 0)
                 return r;
 
+        if (endswith(c, "/"))
+                fn = strappend(c, "user-dirs.dirs");
+        else
+                fn = strjoin(c, "/", "user-dirs.dirs", NULL);
+        if (!fn)
```

Comment:
Indeed, I totally forgot to run valgrind before sending this patch, shame on me.

Not being able to find `strappenda()` anywhere in this project, I instead went for `strappend()` and your `_cleanup_free_` macro as suggested in your coding style.